### ISSUE TITLE
Bad logic when changing subnets on elb

### ISSUE THREAD

**Issue by jjshoe** [hashicorp/terraform#7990]
### Terraform Version

0.6.16
### Affected Resource(s)

Please list the resources as a list, for example:
- aws_elb
### What I see in output

 subnets.2148085103: "subnet-a" => ""
  subnets.3073796272: "" => "subnet-b"
  subnets.3952609156: "subnet-c" => ""
  subnets.862181705:  "" => "subnet-d"
### Error

Failure adding ELB subnets: InvalidConfigurationRequest: ELB cannot be attached to multiple subnets in the same AZ.
    status code: 409
### Bug

It deletes subnet a, so far so good.
It tries to add subnet b, this fails, because it hasn't removed subnet c, which is in the same AZ.
### Workaround

Apply the nuclear approach and delete the ELB.

**Comment by iceycake** [comment]
I have the same issue with 0.7.0 as well.

**Comment by jwadolowski** [comment]
Same here, terraform 0.7.

Resource change:

```
resource "aws_elb" "xyz-dev-front-elb" {
  subnets = [
    # "${aws_subnet.xyz-dev-front-subnet.*.id}",
    "${aws_subnet.xyz-dev-sec-subnet.*.id}",
  ]
}
```

Plan:

```
~ aws_elb.xyz-dev-front-elb
    subnets.#:          "1" => "2"
    subnets.2754519650: "subnet-799d5ax9" => ""
    subnets.2773544518: "" => "subnet-b15945e8"
    subnets.698181429:  "" => "subnet-f991858f"
```

Error after `terraform apply`:

```
aws_elb.xyz-dev-front-elb: Modifying...
  subnets.#:          "1" => "2"
  subnets.2754519650: "subnet-799d5ax9" => ""
  subnets.2773544518: "" => "subnet-b15945e8"
  subnets.698181429:  "" => "subnet-f991858f"
Error applying plan:

1 error(s) occurred:

* aws_elb.xyz-dev-front-preview-elb: Failure adding ELB subnets: InvalidConfigurationRequest: ELB cannot be attached to multiple subnets in the same AZ.
	status code: 409, request id: 795fbc97-686d-11e6-b36e-0bc5d86df6e5
```

**Comment by joshdurbin** [comment]
Same here, Terraform v0.7.4.

~ aws_elb.internal-hapi-dt
    subnets.1880314680: "subnet-06dcd862" => ""
    subnets.3033330828: "subnet-35f8a743" => ""
    subnets.4010762145: "" => "subnet-32f8a744"
    subnets.4204839735: "" => "subnet-00dcd864"


Plan: 0 to add, 1 to change, 0 to destroy.

.............

aws_elb.internal-hapi-dt: Modifying...
  subnets.1880314680: "subnet-06dcd862" => ""
  subnets.3033330828: "subnet-35f8a743" => ""
  subnets.4010762145: "" => "subnet-32f8a744"
  subnets.4204839735: "" => "subnet-00dcd864"
Error applying plan:

1 error(s) occurred:

* aws_elb.internal-hapi-dt: Failure adding ELB subnets: InvalidConfigurationRequest: ELB cannot be attached to multiple subnets in the same AZ.
	status code: 409, request id: 66912dcf-8a86-11e6-93ba-6b9d9958d063

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.

**Comment by catsby** [comment]
Hello – this was fixed in #9131 , thanks!
